src/background_music/ConfigValue.h

```diff
@@ -0,0 +1,130 @@
+#ifndef AUDACIOUS_PLUGINS_CONFIGVALUE_H
+#define AUDACIOUS_PLUGINS_CONFIGVALUE_H
+/*
+ * audacious-plugins/ConfigValue.h
+ *
+ * Added by michel on 2023-09-23
+ * Copyright (C) 2015-2023 Michel Fleur.
+ * Source https://github.com/emmef/audacious-plugins
+ * Email audacious-plugins@emmef.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <algorithm>
+#include <cstddef>
+#include <libaudcore/runtime.h>
+#include <limits>
+#include <stdexcept>
+#include <type_traits>
+
+template<typename T>
+struct ConfigValue
```

Comment:
I would prefer not to have this class added, it's a lot of boilerplate what's effectively just a wrapper around the existing libaudcore API. While I appreciate the effort put into designing it, I'd prefer to see the existing API used directly for consistency with the other plugins.